Base sync

williamspacefire opened the pull request:
<!-- You can remove this first section if you have contributed before -->
### First time contributor checklist
<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [ ] I have read [how to contribute](https://github.com/signalapp/Signal-Android/blob/master/CONTRIBUTING.md) to this project
- [ ] I have signed the [Contributor License Agreement](https://whispersystems.org/cla/)

### Contributor checklist
<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [ ] I am following the [Code Style Guidelines](https://github.com/signalapp/Signal-Android/wiki/Code-Style-Guidelines)
- [ ] I have tested my contribution on these devices:
 * Device A, Android X.Y.Z
 * Device B, Android Z.Y
 * Virtual device W, Android Y.Y.Z
- [ ] My contribution is fully baked and ready to be merged as is
- [ ] I ensure that all the open issues my contribution fixes are mentioned in the commit message of my first commit using the `Fixes #1234` [syntax](https://help.github.com/articles/closing-issues-via-commit-messages/)

----------

### Description
<!--
Describe briefly what your pull request proposes to fix. Especially if you have more than one commit, it is helpful to give a summary of what your contribution as a whole is trying to solve.
Also, please describe shortly how you tested that your fix actually works.
-->

fumiakiy committed:
Check if the content uri already exists and rename the file until it's valid to insert.

Fixes #10159

alex-signal committed:
Apply proper rotation to buttons and video in landscape.

greyson-signal committed:
Force custom emojis for about views.

AsamK committed:
Configure keep alive duration for okhttp connection pool to 1 minute.

The signal http server supports http keep alive, but closes idle
connections after 1 minute.
The default OkHttp connection pool will keep idle connections in the pool
for 5 minutes and doesn't notice it when the server closes connections.
As currently the automatic okhttp retries are disabled, reusing such a
stale connection will be fatal.

Issue is especially severe for incoming calls, which fail because the request
to retrieve the turn servers fails and isn't retried: #10787

greyson-signal committed:
Fix issue where reaction shade is offset in chat bubbles.

Fixes #10843

alan-signal committed:
Fix unnecessary zeros padding.

cody-signal committed:
Allow contact support from registration lock and screen lock screens.

alan-signal committed:
Add additional logging for conflict resolution.

fumiakiy committed:
Adds some breathing space in the button caption.

Fix to a bug reported in the [beta forum](https://community.signalusers.org/t/beta-feedback-for-the-upcoming-android-5-3-release/25088/353)

greyson-signal committed:
Fix storage sync issue related to duplicate remote contacts.

The theory is that if multiple remote keys map to the *same* local
entry, then when we go to update the local contact the second time, we
won't find the entry by StorageID, because we changed it during the
*first*  update, which will then lead to a crash.

This change makes it so dupes are considered invalid, so we'll delete
them and upload our own local copy.

greyson-signal committed:
Clean up unnecessary GCM stuff, improve FCM logging.

moxie0 committed:
Add support for configuring a signal proxy.

alan-signal committed:
Fix migration of null titled group.

alan-signal committed:
Skip automigration of nameless groups.

cody-signal committed:
Fix empty conversation update item text.

For some reason, if an EmojiTextView has a wrap content width and some other set of conditions occur, the view will not request a relayout when text changes.
This change inelegantly calls request layout more often to prevent that from happening.

greyson-signal committed:
Add UI support for configuring a proxy.

greyson-signal committed:
Allow using a proxy during registration.

greyson-signal committed:
Updated language translations.

greyson-signal committed:
Bump version to 5.3.11

greyson-signal committed:
Remove conversation update min width.

alan-signal committed:
Do not linkify message body if recipient is not message request accepted.


Co-authored-by: Greyson Parrelli <greyson@signal.org>

AsamK committed:
Fix crash when changing contact color without contacts permission.

On mulit-device app, after the contact color is changed, a multi-device
contact update job is triggered, which tries to access the system
avatar.
This causes a crash if the user has revoked the contacts permission.

greyson-signal committed:
Add additional debug info for internal users.

greyson-signal committed:
Clean up several UX interactions with proxy entry.

greyson-signal committed:
Add support for sgnl:// proxy deep links.

greyson-signal committed:
Allow proxy deep links during registration.

greyson-signal committed:
Use a simple check to verify proxies during registration.

alan-signal committed:
Use matching color for read conversation sender names.

greyson-signal committed:
Improve proxy link parsing.

greyson-signal committed:
Change migration prompt from 'update' to 'upgrade'.

greyson-signal committed:
Add some description to the proxy settings screen.

greyson-signal committed:
Increase thread string length to 5 in logs.

greyson-signal committed:
Updated language translations.

greyson-signal committed:
Bump version to 5.3.12

AsamK committed:
Fix crash from incoming call without contacts permission.

When the device is in Do Not Disturb mode and a call comes in from a
system contact, the app crashes if the user has revoked the contacts
permission.
The crash occurs because in Do Not Disturb mode Signal tries to check if
the contact is starred.

greyson-signal committed:
Prevent narrow race condition when resetting network components.

greyson-signal committed:
Fix success dialog in proxy edit screen.

greyson-signal committed:
Include git hash in debuglog.

jack-signal committed:
Bump libsignal-client to 0.2.3

alex-signal committed:
Dismiss reactions when read on linked devices.

Also sends out read receipts for read reactions.

At present, only iPad is sending these -- desktop still needs to add send support.

greyson-signal committed:
Add transparency to the compose bar when wallpaper is present.

AsamK committed:
Adapt maxInstancesForQueue to only consider instances of the same job.

Currently the maxInstancesForQueue limit checks the count of all jobs in a
given queue. If there are already too many jobs, the new job is discarded.

However this is not the expected behavior for the two jobs where it's used:
GroupCallPeekWorkerJob and AutomaticSessionResetJob
For both the expected behavior is that there aren't too many jobs of them
started, but that there will be at least one instance of them started.
Both of them use the same queue as the PushProcessMessageJob and the MarkerJob.
Those two jobs are often in the queue at the same time, effectively preventing
the GroupCallPeekWorkerJob and AutomaticSessionResetJob from being enqueued.

fumiakiy committed:
Fix reaction overlay not showing on first try in RTL mode.

Moved the code that does some layouts according to the view's layout direction to the place where we can safely use the value from getLayoutDirection().

This fixes the issue reported in [the beta 5.3 forum](https://community.signalusers.org/t/beta-feedback-for-the-upcoming-android-5-3-release/25088/315).

greyson-signal committed:
Fix text overlapping on edit proxy screen.

greyson-signal committed:
Add a system to improve app foreground observation.

There was previously a crash that occurred when multiple threads tried to use ProcessLifecycleOwner, and this will hopefully resolve that.

greyson-signal committed:
Updated image compression parameters.

aimiebell commented on app/src/main/java/org/thoughtcrime/securesms/conversation/ConversationAdapter.java:
```diff
@@ -112,6 +112,7 @@
   private boolean             hasWallpaper;
+  private boolean             isMessageRequestAccepted;
 
   ConversationAdapter(@NonNull LifecycleOwner lifecycleOwner,
```
I detect that this code is problematic. According to the [Bad practice (BAD_PRACTICE)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#bad-practice-bad-practice), [Nm: Method names should start with a lower case letter (NM_METHOD_NAMING_CONVENTION)](https://spotbugs.readthedocs.io/en/stable/bugDescriptions.html#nm-method-names-should-start-with-a-lower-case-letter-nm-method-naming-convention).
Methods should be verbs, in mixed case with the first letter lowercase, with the first letter of each internal word capitalized.

cody-signal committed:
Fix timestamp and size for attachment saves.

greyson-signal committed:
Switch to a standard toolbar color.

* Switch to a standard toolbar color.

* T r a n s p a r e n c y

* Add back a toolbar shadow.

* Some more theming

* Also fix verify identity screen.

* Hide shadow on empty conversations.

* Slightly less transparent.

greyson-signal committed:
Disable conversation shortcuts when screen lock is enabled.

alan-signal committed:
Fix missing UUID crash, get group recipients direct from V2 group record.

greyson-signal committed:
Fix issue where group sent transcripts were not put in proper processing queue.

greyson-signal committed:
Fail an enqueued job if its dependencies already failed.

This was a bug that was most notable during the attachment pre-upload
process: if an attachment failed to upload, the subsequently-enqueued
PushMediaSendJob would still send. This is because the attachment jobs
were enqueued first and failed *before* we enqueued the PushMediaSendJob
as a dependency.

This will use the JobTracker to determine if a dependency already failed
at the time of enqueueing a job like this. This isn't perfect, because
the JobTracker is memory-only and has a limited buffer (currently 1000),
but in practice this should be sufficient for our use cases. I imagine
it'd only fall apart if we somehow  enqueued a dependent job *much*
later, or somehow enqueued it based on a job ID that we persisted on
disk through an app restart. We don't do any of these things, currently,
and probably never should.

Also took the opportunity to patch a case where we weren't failing
dependent jobs when canceling a job, since I was giving the failure
stuff a look-over.

greyson-signal committed:
Ensure we only have one IncomingMessageObserver.

We saw a worrisome log that implied there may be a situation where
there's two IncomingMessageObservers. I can't see how that would happen,
but this is a failsafe to prevent that from happening.

greyson-signal committed:
Remove unused signaling key code.

alan-signal committed:
Periodic alarm to check for messages.

cody-signal committed:
Updated language translations.

cody-signal committed:
Bump version to 5.4.0

greyson-signal committed:
Fix banner position issues.

greyson-signal committed:
Lighten and unify toolbar shadows.

greyson-signal committed:
Fix NPE in AppForegroundObserver.

greyson-signal committed:
Updated language translations.

greyson-signal committed:
Bump version to 5.4.1

greyson-signal committed:
Fix landscape boundaries of conversation activity.

greyson-signal committed:
Fix boundaries of conversation banners in landscape.

cody-signal committed:
Fix crash when running shortcut update job on older APIs.

alan-signal committed:
Shorten message processor foreground service delay to resolve ANR.

greyson-signal committed:
Do not use View.getLayoutDirection(). 

This value doesn't populate until after the first layout pass. Instead,
it appears to be safer to just read it from the Configuration.

greyson-signal committed:
Archive sessions on 409/410 instead of deleting them.

greyson-signal committed:
Update long message activity toolbar color.

greyson-signal committed:
Ensure a job has a context before we fail it.

cody-signal committed:
Fix RTL padding issue for conversation items.

greyson-signal committed:
Update chat wallpaper preview toolbar colors.

greyson-signal committed:
Bump Lottie version to 3.6.0

Should hopefully fix an NPE we're seeing.

greyson-signal committed:
Move BlobProvider storage out of cache and into internal storage.

greyson-signal committed:
Improve network reliability.

greyson-signal committed:
Make max image dimensions a round number.

cody-signal committed:
Updated language translations.

cody-signal committed:
Bump version to 5.4.2

greyson-signal committed:
Fix conversation banner layout issues.

alan-signal committed:
Increase maximum zoom level for small images.

alan-signal committed:
Ensure MessageProcessReceiver pending result finished is called timely.

cody-signal committed:
Do not show message notifications if disabled in settings.

greyson-signal committed:
Improve experience of populating migration dialog.

greyson-signal committed:
Fix status bar color on Android 5.x.

cody-signal committed:
Revert "Bump libsignal-client to 0.2.3"

This reverts commit 8b7506ed2defbf71333b5077852630a2fa11d088.

alan-signal committed:
Force disabling of foreground notification vibration.

cody-signal committed:
Updated language translations.

cody-signal committed:
Bump version to 5.4.3

alan-signal committed:
Upgrade notification channels earlier.

alan-signal committed:
Bump version to 5.4.4
